### PR TITLE
 LLDB support for "Extend ClangImporter to import clang modules from DWARF

### PR DESCRIPTION
**Pull request by adrian-prantl**
see https://github.com/apple/swift/pull/20600.

### CODE REVIEW

File: packages/Python/lldbsuite/test/lang/swift/dwarfimporter/TestSwiftDWARFImporter.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        # Remove the header files to thwart ClangImporter.
+        self.assertTrue(os.path.isdir(include))
+        shutil.rmtree(include)
+        target, process, thread, bkpt = lldbutil.run_to_source_breakpoint(
+            self, 'break here', lldb.SBFileSpec('main.swift'))
+        frame = thread.frames[0]
+        lldbutil.check_variable(self, frame.FindVariable("pureSwift"),
+                                True, value="42")
+
+if __name__ == '__main__':
```
Comment (dcci) [inline packages/Python/lldbsuite/test/lang/swift/dwarfimporter/TestSwiftDWARFImporter.py:47]: do you need this?

File: packages/Python/lldbsuite/test/lang/swift/dwarfimporter/TestSwiftDWARFImporter.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# TestSwiftDWARFImporter.py
```
Comment (dcci) [inline packages/Python/lldbsuite/test/lang/swift/dwarfimporter/TestSwiftDWARFImporter.py:1]: I'm leaning towards "let's remove all the copyrights" for tests, here. Also, the year is outdated :)

**Review by dcci — COMMENTED** [review]
This is really cool :)

**Comment by adrian-prantl** [comment]
test with https://github.com/apple/swift/pull/20600
@swift-ci test

**Comment by adrian-prantl** [comment]
test with https://github.com/apple/swift/pull/20600
@swift-ci test

**Comment by adrian-prantl** [comment]
test with https://github.com/apple/swift/pull/20600
@swift-ci test

File: source/Core/ModuleList.cpp [hunk 3]
```diff
@@ -74,11 +74,18 @@ static constexpr PropertyDefinition g_properties[] = {
...
+    {"use-swift-dwarfimporter", OptionValue::eTypeBoolean, false, true, nullptr,
+     {}, "Reconstruct Clang module dependencies from DWARF "
+         "when debugging Swift code"},
     {"clang-modules-cache-path", OptionValue::eTypeFileSpec, true, 0, nullptr,
      {},
      "The path to the clang modules cache directory (-fmodules-cache-path)."}};
 
-enum { ePropertyEnableExternalLookup, ePropertyClangModulesCachePath };
+enum {
+  ePropertyEnableExternalLookup,
```
Comment (dcci) [inline source/Core/ModuleList.cpp:85]: These enum values are a little opaque. Can you add a comment?

File: source/Core/ModuleList.cpp [hunk 4]
```diff
@@ -104,6 +111,12 @@ FileSpec ModuleListProperties::GetClangModulesCachePath() const {
       ->GetCurrentValue();
 }
 
+bool ModuleListProperties::GetUseDWARFImporter() const {
+  const uint32_t idx = ePropertyUseDWARFImporter;
+  return m_collection_sp->GetPropertyAtIndexAsBoolean(
```
Comment (dcci) [inline source/Core/ModuleList.cpp:116]: you can probably fold `idx` in the caller.

**Review by dcci — APPROVED** [review]

File: source/Core/ModuleList.cpp (hunk 4 above)
Comment (adrian-prantl) [reply source/Core/ModuleList.cpp:116]: This is a pattern in LLDB. Doing it the same way as everywhere will make it easier to search&replace these accessors once we come up with a nicer templated schema.

File: source/Core/ModuleList.cpp (hunk 3 above)
Comment (adrian-prantl) [reply source/Core/ModuleList.cpp:85]: Same here, it's a pattern. We don't document property enums where they are declared, the documentation is in the struct above. Eventually we'll redo how properties are implemented and consolidate everything in a less hand-written fashion. See tha (later) comments in https://reviews.llvm.org/D43647.

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
test with https://github.com/apple/swift/pull/20600
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test and merge

**Pull request merged**